src/clib/pioc.c

```diff
@@ -718,6 +718,17 @@ int PIOc_InitDecomp(int iosysid, int pio_type, int ndims, const int *gdimlen, in
     }
     *ioidp = pio_add_to_iodesc_list(iodesc, comm);
 
+    /* Check whether we have exceeded the maximum number of ioids (PIO_IODESC_MAX_IDS).
+     * This limit is necessary since each file uses a sparse pointer array (with a fixed
+     * size of PIO_IODESC_MAX_IDS) to look up a data buffer per ioid.
+     * FIXME: Replace the sparse array with a map or hash map to get rid of this limit
+     */
+    if (*ioidp - PIO_IODESC_START_ID + 1 > PIO_IODESC_MAX_IDS)
```

Comment:
Ideally this error check should go in pio_add_to_iodesc_list() where the id is assigned first to the iodesc . However the function signature for pio_add_to_iodesc_list() does not return an error, so it gets tricky unless we change the function signature. What do you think, should we change the function signature or just catch the error here?

Reply:
Since this is only a short-term fix, I think it is OK to check the error in PIOc_InitDecomp for now.

Reply:
ok, thanks